docs/concepts-advanced/environment-variables.md

```diff
@@ -1,170 +1,352 @@
+!!! info
+    If environment files aren't loading, confirm that you're using Lagoon base images, that you're copying them into the contianer as part of your `Dockerfile`, and that your container has a `WORKDIR` setting that points to the location of these files.
 
-The syntax in the environment files is as following:
+The syntax of an environment file is `ENV_VAR_NAME="env var value"`:
 
-```bash title="myenvironment.env"
-MYVARIABLENAME="MyVariableValue"
-MVARIABLENUMBER=4242
+```bash title=".env"
+PHP_MEMORY_LIMIT="400M"
+PHP_MAX_EXECUTION_TIME=900
 DB_USER=$DB_USERNAME # Redefine DB_USER with the value of DB_USERNAME e.g. if your application expects another variable name for the Lagoon-provided variables.
 ```
 
-### `.lagoon.env.$BRANCHNAME`
+### `.lagoon.env` and `.lagoon.env.BRANCHNAME`
+
+Use `.lagoon.env` when you want to always set env vars, but only for Lagoon images.
 
-If you want to define environment variables different per environment you can create a `.lagoon.env.$BRANCHNAME` e.g. for the main branch `.lagoon.env.main`. This helps you keep environment variables apart between environments.
+Use `.lagoon.env.BRANCHNAME` when you want to set different env vars per environment. `BRANCHNAME` will be replaced by the env vars `$LAGOON_GIT_BRANCH` or `$LAGOON_GIT_SAFE_BRANCH`. For example, if you deployed the branch `main`, the file name would be `.lagoon.env.main`. Branch names that contain special characters are sanitized, so check the value of `$LAGOON_GIT_SAFE_BRANCH` to determine what the filename should be.
 
 ### `.env` and `.env.defaults`
 
-`.env` and `.env.defaults` will act as the default values for environment variables if none other is defined. For example, as default environment variables for pull request environments (see [Workflows](workflows.md#pull-requests)).
+Use `.env` and `.env.defaults` when you want to always set env vars, including for local development and non-Lagoon tools that also read them.
+
+## Docker Environment Variables
+
+### Docker Images
+
+Env vars can be set in docker images by using the [`ENV` command](https://docs.docker.com/reference/dockerfile/#env) in your `Dockerfile`. These will be set when Lagoon builds your projects images and are available at runtime.
+
+### Docker Compose
+
+Lagoon doesn't use env vars set using Docker Compose when deploying your projects. You may still choose to set them for local development purposes. Review the [Docker Compose env var documentation](https://docs.docker.com/compose/how-tos/environment-variables/) for more information.
 
-## Special Environment Variables
 
-### `PHP_ERROR_REPORTING`
+## Build Variables
+
+Use build vars when you want to set an env var that is valid at buildtime, but only for a single deployment. Bulk deployments also support build vars.
```

Comment:
With this, it might also make sense to add an example use case to the documentation.

Reply:
in fact, just explaining what we're doing in the example would be great